src/mbgl/renderer/paint_parameters.cpp

```diff
@@ -103,53 +103,71 @@ gfx::DepthMode PaintParameters::depthModeFor3D() const {
     return gfx::DepthMode{gfx::DepthFunctionType::LessEqual, gfx::DepthMaskType::ReadWrite, {0.0, depthRangeSize}};
 }
 
-void PaintParameters::clearStencil() {
-    nextStencilID = 1;
-    tileClippingMaskIDs.clear();
-
-#if MLN_RENDER_BACKEND_METAL
-    // Metal doesn't have an equivalent of `glClear`, so we clear the buffer by drawing zero to (0:0,0)
-    std::set<UnwrappedTileID> ids{{0, 0, 0}};
-    auto f = [](const auto& ii) -> const UnwrappedTileID& {
-        return ii;
-    };
-    renderTileClippingMasks(ids.cbegin(), ids.cend(), std::move(f), /*clear=*/true);
-#else
-    context.clearStencilBuffer(0b00000000);
-#endif
-}
-
 namespace {
 
 template <typename TIter>
 using GetTileIDFunc = std::function<const UnwrappedTileID&(const typename TIter::value_type&)>;
```

Comment:
Not related to this PR but I think this a function pointer would work as well here and would be more performant.